### PR TITLE
refactor: split the Python source-root helpers (#1669)

### PR DESCRIPTION
**Pull request by vitali87**
## Summary

Two functions in `codebase_rag/parsers/python_source_roots.py` carry SonarCloud `python:S3776` findings on `main` (`_package_dir_remaps` and `resolve_via_source_roots`). Both are split into named helpers with no behaviour change.

- `_package_dir_remaps` now reads the table through `_setuptools_package_dir` (the `[tool.setuptools.package-dir]` lookup, empty for a malformed file or a non-table at any level), resolves each value through `_remap_target` (non-string, non-directory and repo-escaping values all yield nothing), and expands the default `"" = "lib"` key through `_default_remap_children`.
- `resolve_via_source_roots` now collects candidates through `_matching_roots` (longest import prefix first), tests each against disk through `_exists_on_disk`, and builds the answer through `_dotted_module_path`.
- One redundant condition is dropped: the old on-disk test for a non-empty rest also checked `(sub / __init__.py).is_file()`, which cannot be true unless `sub.is_dir()` is already true, since `is_file()` requires every parent component to be a traversable directory.

Public signatures are unchanged. The only production callers are in `import_processor.py`.

`discover_python_source_roots` in the same file is also over the threshold and is deliberately left alone: #1669 tracks these as per-function follow-ups, and mixing them would make the behaviour-preservation argument harder to check.

## Type of Change

- [x] Refactor (no functional change)

## Related Issues

Part of #1669.

## Test Plan

- [x] New `codebase_rag/tests/test_python_source_roots_units.py`: 22 tests calling the two helpers directly, one per branch, written and run against the UNREFACTORED code first so they pin current behaviour rather than the new shape. Covers a malformed pyproject, a non-table at each of the three key levels, a non-string value, a missing directory, a file target, a repo-escaping `..` target, named and dotted remap keys, the default-key expansion (with `__init__.py` and non-Python children excluded), an unknown top-level name, a prefix that matches without a dot boundary, exact matches on a package and on a single module, submodule confirmation by file and by directory, disambiguation between same-named roots, a trusted sole match, several unconfirmed matches, and longest-prefix ordering.
- [x] 34 tests green (the 22 new plus the 12 pre-existing end-to-end ones in `test_python_source_root_imports.py`).
- [x] 12 mutations, each reddening only its own tests: a repo-escaping remap kept; a file target accepted; a non-string value coerced with `str()`; `__init__.py` counted as a child; a named remap expanded like the default one; a non-table package-dir iterated; the prefix dot boundary dropped; longest-prefix ordering reversed; module files not confirming a match; a sole match not trusted; any unconfirmed match trusted; the rest not joined onto the dotted directory.
- [x] The local reviewer ran the old and new code differentially over 124 remap shapes and 924 resolution cases with zero differences, each harness validated with planted mutations first.
- [x] The pre-commit `ty` hook is red at this machine's baseline (identical on `main`, nothing in a touched file) and was skipped for the commits; every other hook ran. `ty check` on the changed module passes.

## Review note, carried from the local round

The first version of the non-string-value test used `bad = 3` with no directory named `3`, so it passed whether or not the `isinstance` guard existed: `str(3)` resolved to a path that did not exist and was rejected by the later directory check. The fixture now creates that directory, so only the guard can reject the value, and the matching mutation uses the exact coercion that survived before.

The review bot asked for a more descriptive name than the original `_joined`, matching the repository's naming rule; it is now `_dotted_module_path` with a docstring stating what it returns.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Python source-root resolution for package and module paths.
  * Added safer handling for malformed configuration, invalid paths, and repository-boundary cases.
  * Preserved consistent resolution behavior across varied package layouts.

* **Tests**
  * Added comprehensive coverage for source-root mappings, dotted package names, path validation, and resolution precedence.

### CODE REVIEW

**Comment by vitali87** [comment]
@greptileai review

File: codebase_rag/parsers/python_source_roots.py [hunk 1]
```diff
@@ -109,6 +129,35 @@ def _add(name: str, dotted_dir: str) -> None:
...
+        if module_name == prefix:
+            matches.append(("", dotted_dir))
+        elif module_name.startswith(prefix + cs.SEPARATOR_DOT):
+            matches.append((module_name[len(prefix) + 1 :], dotted_dir))
+    matches.sort(key=lambda m: len(m[0]))
+    return matches
+
+
+def _joined(dotted_dir: str, rest: str) -> str:
+    return f"{dotted_dir}{cs.SEPARATOR_DOT}{rest}" if rest else dotted_dir
```
[the remark this answers is not in the document]
Comment (vitali87) [reply codebase_rag/parsers/python_source_roots.py]: Renamed to `_dotted_module_path`, with a docstring saying what it returns: the dotted qualified name for `rest` inside the root at `dotted_dir`. All three call sites updated; no other module referenced it.

**Comment by vitali87** [comment]
@greptileai review
